src/g_text.c

```diff
@@ -811,12 +817,11 @@ void gatom_key(void *z, t_symbol *keysym, t_floatarg f)
             rtext_key(t, '.', &s_);
             rtext_key(t, 0, gensym("Home"));
         }
-            /* automatically escape special characters in symbols */
-        if (x->a_flavor == A_SYMBOL && (c == ' ' || c == ',' || c == ';' ||
-            c == '$' || c == '\\'))
-                rtext_key(t, '\\', &s_);
-            /* and at last, insert the character */
-        rtext_key(t, c, keysym);
+        if (x->a_flavor == A_SYMBOL || x->a_flavor == A_LIST)
+            rtext_key(t, c, keysym);  /* insert the character */
+        else if (x->a_flavor == A_FLOAT && ((c >= '0' && c <= '9') || c == '.' ||
+            c == '-' || c == '+' || c == 'e' || c == 'E' || c == '\b'))
+                rtext_key(t, c, keysym);  /* insert the accepted characters */
```

Comment:
while this is not a regression, i wonder whether we shouldn't use a validator function (`strtod()`) instead of filtering by character.
e.g. the above (and *almost* Pd<0.52) will happily allow the user to type `1.2-3.4+5` which then ends up as *1.2*

Reply:
it was more like a general suggestion. for *this* PR i think it is ok to restore the original behaviour.